astropy/coordinates/baseframe.py

```diff
@@ -1326,8 +1331,10 @@ def _frameattr_equiv(left_fattr, right_fattr):
             else:
                 return False
 
-        if isinstance(left_fattr, BaseCoordinateFrame):
-            if isinstance(right_fattr, BaseCoordinateFrame):
+        left_is_coord = isinstance(left_fattr, BaseCoordinateFrame)
+        right_is_coord = isinstance(right_fattr, BaseCoordinateFrame)
+        if left_is_coord or right_is_coord:
+            if left_is_coord and right_is_coord:
                 # both are coordinates
                 if left_fattr.is_equivalent_frame(right_fattr):
                     return np.all(left_fattr == right_fattr)
```

Comment:
Actually, thinking about how this continues, since we want to make sure the code always returns True or False, I think it would make sense to guard the code at the end in a `try/except`, as in
```
try:
    return np.all(left_attr == right_attr)
except Exception:
    return False
```
With that, in principle, you could have only the both repr and both frame checks, since that would catch all mismatches. But code is probably clearer as you have it!

Reply:
... and catching `Exception` might hide broken stuff. I have a slight preference for the status quo due to that.